client/src/main/java/tw/waterball/survive/io/client/view/GameView.java

```diff
@@ -2,14 +2,15 @@
 
 
 import tw.waterball.survive.io.client.controller.PlayerController;
+import tw.waterball.survive.io.game.gameloop.GameListener;
 import tw.waterball.survive.io.game.sprite.Sprite;
 import tw.waterball.survive.io.game.sprite.role.Role;
 import tw.waterball.survive.io.game.sprite.weapon.Weapon;
 
 import java.util.List;
 
 
-public abstract class GameView {
+public abstract class GameView implements GameListener {
```

Comment:
Put `GameListener` into `GameLoop` as an inner-class, and rename it to `Listener`
so it can be referenced by  `GameLoop.Listener`